File: eBarbershop.Services/Database/Rezervacija.cs
using System;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;

namespace eBarbershop.Services.Database;

public  class Rezervacija
{
    public int RezervacijaId { get; set; }

    public DateTime DatumRezervacije { get; set; }


    public int KorisnikId { get; set; }
    public virtual Korisnik Korisnik { get; set; } = null!;

    public int KlijentId { get; set; }
    public virtual Korisnik Klijent { get; set; } = null!;


    public int UslugaId { get; set; }


    public virtual ICollection<Termin> Termins { get; set; } = new List<Termin>();

    public virtual Usluga Usluga { get; set; } = null!;

}

File: UI/ebarbershop_admin/lib/models/rezervacija.dart
import 'package:ebarbershop_admin/models/korisnik.dart';
import 'package:ebarbershop_admin/models/termin.dart';
import 'package:ebarbershop_admin/models/usluga.dart';
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija.g.dart';

@JsonSerializable()
class Rezervacija {
  int? rezervacijaId;
  DateTime? datumRezervacije;
  int? korisnikId;
  Korisnik? korisnik;
  int? uslugaId;
  Usluga? usluga;
  Termin? termin;
  List<Termin>? termini;
  int? klijentId; // logirani korisnik
  Korisnik? klijent;
  Rezervacija(this.rezervacijaId, this.datumRezervacije, this.korisnikId,
      this.uslugaId, this.usluga, this.korisnik, this.klijentId, this.klijent, this.termin, this.termini);

  factory Rezervacija.fromJson(Map<String, dynamic> json) =>
      _$RezervacijaFromJson(json);
  Map<String, dynamic> toJson() => _$RezervacijaToJson(this);
}
